Release 3.1.3

Nightsd01 opened the pull request:
• Updates to the latest version of the iOS SDK (Android SDK updates are automatically distributed via gradle)
• Adds a new badge handling system in iOS which uses the OneSignalNotificationServiceExtension to perform badge handling logic instead of relying on the backend server
• Removes podfiles which are no longer used to distribute the iOS SDK

Nightsd01 committed:
Release 3.1.3

• Updates to the latest version of the iOS SDK
• Adds a new badge handling system in iOS which uses the OneSignalNotificationServiceExtension to perform badge handling logic instead of relying on the backend server
• Removes podfiles which are no longer used to distribute the iOS SDK

jkasten2 commented:
### iOS
Lets add a link to the App Group setup in the README that will be need for the badge inc feature in the future.

### Android
Also lets update this file to fix the Android decencies 
https://github.com/geektimecoil/react-native-onesignal/blob/3.1.2/android/build.gradle

```gradle
buildscript {
    repositories {
        jcenter()
        mavenCentral()
        maven { url 'https://maven.google.com' }
    }

    dependencies {
        classpath 'com.android.tools.build:gradle:3.0.1'
    }
}

apply plugin: 'com.android.library'

android {
    compileSdkVersion 27
    buildToolsVersion '27.0.3'

    defaultConfig {
        minSdkVersion 15
        targetSdkVersion 27
        versionCode 1
        versionName '1.0'
        // Do not set onesignal_app_id here, it will be set in the app project.
        // Leave onesignal_google_project_number as it is now pulled from the dashboard.
        manifestPlaceholders = [onesignal_app_id: "",
                                onesignal_google_project_number: "REMOTE"]
    }
    buildTypes {
        release {
            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
        }
    }
}

dependencies {
    compile fileTree(include: ['*.jar'], dir: 'libs')
    compile 'com.facebook.react:react-native:+'
    testCompile 'junit:junit:4.12'
    compile 'com.onesignal:OneSignal:3.+'
}

// Add the following to the top (Line 1) of your app/build.gradle if you run into any issues with duplicate classes.
// Such as the following error
//   Error: more than one library with package name 'com.google.android.gms.license'
/*

plugins {
    id 'com.onesignal.androidsdk.onesignal-gradle-plugin' version '0.8.1'
}
apply plugin: 'com.onesignal.androidsdk.onesignal-gradle-plugin'

*/
```

@Nightsd01 Please test this `android/build.gradle` however just to be sure I didn't miss anything.

Let's also add the onesignal-gradle-plugin to the "FAQ / Repeating Issues" section noting the `Error: more than one library with package name 'com.google.android.gms.license'` error as well so it is easy to find this solution.
```gradle
plugins {
    id 'com.onesignal.androidsdk.onesignal-gradle-plugin' version '0.8.1'
}
apply plugin: 'com.onesignal.androidsdk.onesignal-gradle-plugin'
```

Nightsd01 committed:
Update README.md

• Adds instructions for iOS to add app groups (needed for badge increment/decrement logic)

Nightsd01 committed:
Updated README.md

• Added details on how to deal with the 'multiple library' error for Android

Nightsd01 committed:
Update Android

• Updates react-native-onesignal target SDK and gradle versions
• Adds the onesignal-gradle-plugin to fix dependency issues in the demo project
